Objects/typeobject.c

```diff
@@ -3947,7 +3951,7 @@ type_setattro(PyTypeObject *type, PyObject *name, PyObject *value)
     if (type->tp_flags & Py_TPFLAGS_IMMUTABLETYPE) {
         PyErr_Format(
             PyExc_TypeError,
-            "can't set attributes of built-in/extension type '%s'",
+            "cannot set attributes of immutable type '%s'",
```

Comment:
Can you please put the attribute name in the error message?

Reply:
Yes

Reply:
Hm, it should be %R, not %r